applications/tari_validator_node/src/p2p/services/mempool/service.rs

```diff
@@ -53,43 +60,51 @@ impl MempoolService {
         mempool_requests: mpsc::Receiver<MempoolRequest>,
         outbound: OutboundMessaging,
         tx_valid_transactions: broadcast::Sender<(Transaction, ShardId)>,
+        epoch_manager: EpochManagerHandle,
+        node_identity: Arc<NodeIdentity>,
     ) -> Self {
         Self {
             transactions: Arc::new(Mutex::new(Vec::new())),
             new_transactions,
             mempool_requests,
             outbound,
             tx_valid_transactions,
+            epoch_manager,
+            node_identity,
         }
     }
 
-    pub async fn run(mut self) {
+    pub async fn run(mut self) -> Result<(), MempoolError> {
         loop {
             tokio::select! {
-                Some(req) = self.mempool_requests.recv() => self.handle_request(req).await,
-                Some(tx) = self.new_transactions.recv() => self.handle_new_transaction(tx).await,
+                Some(req) = self.mempool_requests.recv() => self.handle_request(req).await?,
+                Some(tx) = self.new_transactions.recv() => self.handle_new_transaction(tx).await?,
 
                 else => {
                     info!(target: LOG_TARGET, "Mempool service shutting down");
                     break;
                 }
             }
         }
+
+        Ok(())
     }
 
-    async fn handle_request(&mut self, request: MempoolRequest) {
+    async fn handle_request(&mut self, request: MempoolRequest) -> Result<(), MempoolError> {
         match request {
-            MempoolRequest::SubmitTransaction(transaction) => self.handle_new_transaction(*transaction).await,
+            MempoolRequest::SubmitTransaction(transaction) => self.handle_new_transaction(*transaction).await?,
```

Comment:
All errors from these functions should be logged and not propagated to the service run loop

see `hotstuff_service.rs`